src/client/push-client.js

```diff
@@ -201,10 +195,10 @@ export default class PushClient extends EventDispatch {
    * Get the registration of the service worker being used for push.
    *
    * @return {Promise<ServiceWorkerRegistration>} A Promise that
-   *  resolves to either a ServiceWorkerRegistration or to null if none.
+   *  resolves to either a ServiceWorkerRegistration.
    */
   getRegistration() {
-    return this._registrationPromise;
+    return Promise.resolve(this._registration);
```

Comment:
Perhaps just change the API to not return a Promise here? Seems unnecessary to have more async than we need.